balancer/weightedroundrobin/balancer.go

```diff
@@ -595,32 +593,34 @@ func (w *weightedSubConn) updateConnectivityState(cs connectivity.State) connect
 // will cause the backend weight to be treated as the mean of the weights of the
 // other backends. If forScheduler is set to true, this function will emit
 // metrics through the mtrics registry.
-func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, forScheduler bool) (weight float64) {
+func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, recordMetrics bool) float64 {
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
-	weight = 0
-	if forScheduler {
-		defer endpointWeightsHandle.Record(w.metricsRecorder, weight, []string{w.target, w.locality}...)
+	weight := float64(0)
```

Comment:
_Named return values are totally fine_ if you want to keep it.  Just don't do a bare `return`, since it can be confusing (for a minute I actually thought you removed the return value when I saw the bare return, and had to go back to see what was happening).

(But you don't need to initialize them to 0 as in the previous code; Go always initializes everything to zero.)

Reply:
Switched to named return value.